modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/LazyComponentInitializer.java

```diff
@@ -26,4 +29,15 @@
    */
   void initializeComponent(Location location);
 
+  /**
+   * Calling this method guarantees that the requested component from the configuration will be created.
+   * <p/>
+   * The requested component must exists in the configuration.
+   *
+   * @param location the location of the configuration component.
+   * @throws MuleRuntimeException if there's a problem creating the component or the component does not exists.
+   * @return The {@link List} of created component's names
+   */
+  List<String> createComponent(Location location);
```

Comment:
weird. If the method is called createComponent, I expect it to return the component. Plus, why a List? how can this create many components?